core-bundle/src/Translation/DataCollectorTranslator.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Translation;
+
+use Symfony\Component\Translation\DataCollectorTranslator as SymfonyDataCollectorTranslator;
+use Symfony\Component\Translation\TranslatorInterface as LegacyTranslatorInterface;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+class DataCollectorTranslator extends SymfonyDataCollectorTranslator
+{
+    /**
+     * @var TranslatorInterface|LegacyTranslatorInterface
+     */
+    private $translator;
+
+    private $messages = [];
+
+    /**
+     * @param TranslatorInterface $translator
+     */
+    public function __construct($translator)
+    {
+        parent::__construct($translator);
+
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * Gets the translation from Contao’s $GLOBALS['TL_LANG'] array if the message
+     * domain starts with "contao_". The locale parameter is ignored in this case.
+     */
+    public function trans($id, array $parameters = [], $domain = null, $locale = null): string
+    {
+        $translated = $this->translator->trans($id, $parameters, $domain, $locale);
+
+        // Forward to the default translator
+        if (null === $domain || 0 !== strncmp($domain, 'contao_', 7)) {
+            return $translated;
+        }
+
+        $this->collectMessage($GLOBALS['TL_LANGUAGE'] ?? '', (string) $domain, $id, $translated, $parameters);
+
+        return $translated;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function transChoice($id, $number, array $parameters = [], $domain = null, $locale = null): string
```

Comment:
This method can be removed IMHO, as the parent `transChoice()` method does the same thing and additionally calls the `collectMessage()` method.

Reply:
All method call need to be forwarded to the decorated class, we must not call the parent method. We only extend the parent class because the data collector expects that object.